internal/util/generate.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This package is a CLI tool for generating test exposure key data.
+package util
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"log"
+	"math/big"
+	"time"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+)
+
+const (
+	// the length of a diagnosis key, always 16 bytes
+	dkLen               = 16
+	maxTransmissionRisk = 8
+)
+
+func RandIntervalCount() int32 {
+	n, err := rand.Int(rand.Reader, big.NewInt(144))
+	if err != nil {
+		log.Fatalf("rand.Int: %v", err)
+	}
+	return int32(n.Int64() + 1) // valid values are 1-144
+}
+
+func RandomInt(maxValue int) int {
+	n, err := rand.Int(rand.Reader, big.NewInt(int64(maxValue)))
+	if err != nil {
+		log.Fatalf("rand.Int: %v", err)
+	}
+	return int(n.Int64())
+}
+
+func RandomTransmissionRisk() int {
+	return RandomInt(maxTransmissionRisk) + 1
+}
+
+func RandomArrValue(arr []string) string {
+	return arr[RandomInt(len(arr))]
+}
+
```

Comment:
there’s a good bit of overlap w/ this and the export padding

now that has different constraints because we’re trying to use realistic interval data and tranismission risk groupings, so maybe there’s nothing there to reuse.

Reply:
Right, subtly different ways to generate data. I've at the very least gathered some of the extractable util functions that export padding uses into the new util.